lcls-twincat-motion/Library/POUs/Motion/PMPS/FB_StatePMPSEnables.TcPOU

```diff
@@ -143,8 +143,8 @@ ffNoGoal(
 IF bLockBounds THEN
     // Prevent forward/backward motion if the position of the axis is outside of the upper/lower bounds respectively.
     // Prevent forward/backward motion if the command to the axis is outside of the upper/lower bounds respectively.
-    bForwardEnabled  := stMotionStage.Axis.NcToPlc.ActPos < fUpperPos AND stMotionStage.Axis.NcToPlc.TargetPos < fUpperPos;
-    bBackwardEnabled := stMotionStage.Axis.NcToPlc.ActPos > fLowerPos AND stMotionStage.Axis.NcToPlc.TargetPos > fLowerPos;
+    bForwardEnabled  := stMotionStage.Axis.NcToPlc.ActPos < fUpperPos AND stMotionStage.fPosition < fUpperPos;
+    bBackwardEnabled := stMotionStage.Axis.NcToPlc.ActPos > fLowerPos AND stMotionStage.fPosition > fLowerPos;
```

Comment:
This must have been a wild race condition

Reply:
yeah it was...